src/lib/viz/source/__tests__/DatasetSource.spec.ts

```diff
@@ -0,0 +1,180 @@
+      setDefaultCredentials({ ...TEST_CREDENTIALS });
+    });
+
+    it('should use provided credentials', () => {
+      const credentials = new Credentials(
+        'not_default_username',
+        'not_default_apikey',
+        'https://notdefaultserver.com'
+      );
+
+      const source = new DatasetSource(DEFAULT_DATASET, { credentials });
+      expect(source.credentials).toBe(credentials);
+    });
+
+    it('should use default credentials if not provided', () => {
+      const source = new DatasetSource(DEFAULT_DATASET);
+      expect(source.credentials).toBe(defaultCredentials);
+    });
+  });
+
+  describe('Instantiation fails', () => {
+    beforeEach(() => {
+      setDefaultCredentials({ ...TEST_CREDENTIALS });
+
+      Client.prototype.instantiateMapFrom = jest.fn().mockImplementation(() => {
+        throw new Error('Error fake');
+      });
+    });
+
+    it('should fail with error in instantiation', async () => {
+      const source = new DatasetSource(DEFAULT_DATASET);
+
+      expect(async () => {
+        await source.init({ sample: new Set(), aggregation: new Set() });
+      }).rejects.toEqual(new Error('Error fake'));
+    });
+  });
+
+  describe('mapConfig', () => {
+    let instantiateMapFromMock: any;
+
+    beforeEach(() => {
+      setDefaultCredentials({ ...TEST_CREDENTIALS });
+
+      instantiateMapFromMock = jest.fn().mockImplementation(() => {
+        return {
+          metadata: {
+            url: {
+              vector: {
+                urlTemplate: '{s}.carto.com',
+                subdomains: []
+              }
+            },
+            layers: [
+              {
+                meta: {
+                  stats: {
+                    geometryType: 'Point'
+                  }
+                }
+              }
+            ]
+          }
+        };
+      });
+      Client.prototype.instantiateMapFrom = instantiateMapFromMock;
+    });
+
+    it('should have default mapConfig', async () => {
+      const source = new DatasetSource(DEFAULT_DATASET);
+      await source.init({ sample: new Set(), aggregation: new Set() });
+
+      const expectedmapConfig = {
+        vectorExtent: 2048,
+        vectorSimplifyExtent: 2048,
+        bufferSize: {
+          mvt: 10
+        },
+        metadata: {
+          geometryType: true,
+          columnStats: {
+            topCategories: 32768,
+            includeNulls: true
+          },
+          dimensions: true,
+          sample: {
+            num_rows: 1000,
+            include_columns: []
+          }
+        },
+        aggregation: {
+          columns: {},
+          dimensions: {},
+          placement: 'centroid',
+          resolution: 1,
+          threshold: 1
+        },
+        sql: `SELECT * FROM ${DEFAULT_DATASET}`
+      };
+
+      expect(instantiateMapFromMock.mock.calls[0][0]).toMatchObject(expectedmapConfig);
+    });
+
+    it('should not overwrite mapConfig with custom paramters', async () => {
```

Comment:
Should the name be the opposite? "it should allow overwriting mapConfig with custom parameters" ?